solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -1052,7 +1069,39 @@ contract LiquidityProtection is TokenHandler, ContractRegistryClient, Reentrancy
 
         require(averageRateInRange(spotRateN, spotRateD, averageRateN, averageRateD, averageRateMaxDeviation), "ERR_INVALID_RATE");
 
-        return Fraction(averageRateN, averageRateD);
+        return (spotRateN, spotRateD, averageRateN, averageRateD);
+    }
+
+    /**
+      * @dev returns the various rates between the reserves
+      *
+      * @param _poolToken       pool token
+      * @param _reserveToken    reserve token
+      * @param _addSpotRateN    add spot rate numerator
+      * @param _addSpotRateD    add spot rate denominator
+      * @return see `struct PackedRates`
+    */
+    function packRates(
+        IDSToken _poolToken,
+        IERC20Token _reserveToken,
+        uint256 _addSpotRateN,
+        uint256 _addSpotRateD)
+        internal view returns (PackedRates memory)
+    {
+        (uint256 removeSpotRateN, uint256 removeSpotRateD, uint256 removeAverageRateN, uint256 removeAverageRateD) = reserveTokenRates(_poolToken, _reserveToken);
+
+        require(_addSpotRateN <= MAX_UINT128 && _addSpotRateD <= MAX_UINT128, "ERR_INVALID_RATE");
+        require(removeSpotRateN <= MAX_UINT128 && removeSpotRateD <= MAX_UINT128, "ERR_INVALID_RATE");
+        require(removeAverageRateN <= MAX_UINT128 && removeAverageRateD <= MAX_UINT128, "ERR_INVALID_RATE");
```

Comment:
Perhaps combining this into a single `require` will save some gas?

Reply:
It won't be a significant save, but since the revert message is the same for all of the cases - it won't hurt

Reply:
yup, just a bit cleaner and 2 seconds to do